### DISCUSSION TITLE
Localize user-facing messages and parameters

### DISCUSSION THREAD

**Discussion by thibaultcha** [Kong/kong#7205] [Ideas and feature requests]
Provide a way for providers to customize error messages returned to consumers (for example the JWT plugin currently says "Could not find consumer for 'username=bob'", but maybe the provider does not want to expose the notion of "consumers" to them, and configure something like "This JWT could be be verified, please create an account at: ...".

This also overlaps #324 for customizable headers.

**Comment by b6t** [comment]
+1 from gitter How can I customize the response coming from Kong plugins? For example, 

```
{
  "message": "API rate limit exceeded”
}
```

is returned from the rate limiting plugin. Could I reformat this and add info to it like 

```
{
   “status” : false,
    “item": “foo",
    "endpoint": “/foo/bar/bang",
    "request": {
      "eventId": "1540260"
    } ,
   "message": "API rate limit exceeded”
}
```

**Comment by thibaultcha** [comment]
Some thoughts on the requirements:
- should be possible to tweak an HTTP response (currently a JSON response with only a “message" field, but maybe more? What if the responses has 2-3 fields?)
- should implement it in such a way that all responses sent from Kong can be overridden with this system. This enforces consistency (all responses can be tweaked) and limits the amount of work needed by plugins developers
- perhaps also allow to cutomize the status code?
- Iideally the translations/configurations would be in the datastore, to avoid the amount of work needed by providers to localize their responses.

As a first step, simply localizing the `message` field would already be a bonus. Even if not from the datastore.
